### PR TITLE
feat(generator):  add generator vitejs react v1

### PR DESCRIPTION
**Pull request by aboungnaseng-ovhcloud**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` / 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes
| Tickets          | MANAGER-11267
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Add generator V1, generate react application with vitejS

### CODE REVIEW

File: packages/manager/core/ovh-chakra-showcase/package.json [hunk 1]
```diff
@@ -24,6 +24,6 @@
     "@types/react": "^18.0.15",
     "@types/react-dom": "^18.0.6",
     "@vitejs/plugin-react": "^2.0.0",
-    "vite": "^3.0.0"
+    "vite": "^3.0.2"
```
Comment (chipp972) [inline packages/manager/core/ovh-chakra-showcase/package.json:27]: nécessaire ?

File: packages/manager/components/react/breadcrumb/src/Breadcrumb.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/**
+ *  TODO - Move in the future
```
Comment (chipp972) [inline packages/manager/components/react/breadcrumb/src/Breadcrumb.tsx:2]: On pourrait juste le mettre dans le template du generateur de base pour le moment pour éviter d'aujouter un workspace, non ?

File: lerna.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
     "packages/manager/apps/*",
     "packages/manager/core/*",
     "packages/manager/modules/*",
-    "packages/manager/tools/*"
+    "packages/manager/tools/*",
+    "packages/manager/components/react/*"
```
Comment (chipp972) [inline lerna.json:10]: Si on vire Breadcrumb on peut supprimer cette ligne

File: package.json [hunk 4]
```diff
@@ -10,7 +10,8 @@
     "packages/manager/apps/*",
     "packages/manager/core/*",
     "packages/manager/modules/*",
-    "packages/manager/tools/*"
+    "packages/manager/tools/*",
+    "packages/manager/components/react/*"
```
Comment (chipp972) [inline package.json:14]: Si on vire Breadcrumb on peut supprimer cette ligne

File: .eslintignore [hunk 5]
```diff
@@ -2,3 +2,6 @@ dist
 node_modules
 packages/manager/tools/sao-ovh-manager-module/template/src
 packages/manager/tools/sao-ovh-manager-app/template
+packages/manager/apps/container
```
Comment (anooparveti) [inline .eslintignore:5]: **question:** Any particular reason to ignore ESLint on `container`?

File: .eslintrc.js [hunk 6]
```diff
@@ -41,6 +41,7 @@ module.exports = {
         '@typescript-eslint/explicit-module-boundary-types': 'off',
         '@typescript-eslint/no-empty-function': 'off',
         'import/extensions': 'off',
+        'import/prefer-default-export': 'off',
```
Comment (anooparveti) [inline .eslintrc.js:44]: This is widely used as we define constants in a separate file and use default export the defined constants as an object. 
May I know why this is turned off? Any blockers?

File: docs/docs/guide/micro-applications.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
+# Micro-applications
+
+All micro-applications (or uApp) are located in the `packages/manager/apps/*` workspace like other Applications.
+They are powered by :
+
+- [Vite](https://vitejs.dev/) - Next frontend tooling Next Generation Frontend ToolingNext Generation Frontend Tooling
```
Comment (anooparveti) [inline docs/docs/guide/micro-applications.md:6]:
```suggestion
- [Vite](https://vitejs.dev/) -  Next Generation Frontend Tooling
```

File: packages/components/ovh-shell/__tests__/plugin/logger/logger.spec.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+defineFeature(feature, (test) => {
+  let loggerPlugin;
+
+  afterEach(() => {
+    localStorage.removeItem('MANAGER_SHELL_DEBUG');
+  });
+
+  // define i18n instanciation
+  const givenLoggerPluginInstanciated = (given: DefineStepFunction) => {
```
Comment (anooparveti) [inline packages/components/ovh-shell/__tests__/plugin/logger/logger.spec.ts:17]: **nitpick:** Typo
```suggestion
  const givenLoggerPluginInstantiated = (given: DefineStepFunction) => {
```

File: packages/components/ovh-shell/__tests__/plugin/logger/logger.spec.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    given('I have a logger plugin instanciated', () => {
```
Comment (anooparveti) [inline packages/components/ovh-shell/__tests__/plugin/logger/logger.spec.ts:18]:
```suggestion
    given('I have a logger plugin instantiated', () => {
```

File: packages/components/ovh-shell/__tests__/plugin/logger/logger.spec.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+  test('Enable the plugin', ({ given, and, when, then }) => {
+    const objectParam = {
+      test: 'jest',
+      foo: 'bar',
+    };
+
+    const errorParam = new Error('Log error');
+
+    givenLoggerPluginInstanciated(given);
```
Comment (anooparveti) [inline packages/components/ovh-shell/__tests__/plugin/logger/logger.spec.ts:37]:
```suggestion
    givenLoggerPluginInstantiated(given);
```

File: packages/components/ovh-shell/features/plugin/logger/logger.feature [hunk 11]
```diff
@@ -0,0 +1,12 @@
+Feature: logger Plugin
+
+Scenario: Enable the plugin
+  Given I have a logger plugin instanciated
```
Comment (anooparveti) [inline packages/components/ovh-shell/features/plugin/logger/logger.feature:4]:
```suggestion
  Given I have a logger plugin instantiated
```

File: packages/components/ovh-shell/features/plugin/logger/logger.feature [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+  And debug flag is activated
+  When I try to log, info, warn, error, debug something
+  Then I should see the result in console
+
+Scenario: Disable the plugin
+  Given I have a logger plugin instanciated
```
Comment (anooparveti) [inline packages/components/ovh-shell/features/plugin/logger/logger.feature:10]:
```suggestion
  Given I have a logger plugin instantiated
```

File: packages/components/ovh-shell/src/client/index.ts [hunk 13]
```diff
@@ -79,9 +82,9 @@ export default function init(applicationId: ApplicationId) {
   let initPromise;
 
   if (isTopLevelApplication()) {
-    initPromise = fetchApplications().then((apps) =>
-      initStandaloneClientApi(applicationId, apps),
-    );
+    initPromise = fetchApplications().then((apps) => {
+      return initStandaloneClientApi(applicationId, apps);
+    });
```
Comment (anooparveti) [inline packages/components/ovh-shell/src/client/index.ts:87]: **nitpick:** Unnecessary change
```suggestion
    initPromise = fetchApplications().then((apps) =>
     initStandaloneClientApi(applicationId, apps),
    );
```

File: packages/components/ovh-shell/src/client/index.ts [hunk 14]
```diff
@@ -60,9 +60,12 @@ export function initStandaloneClientApi(
...
   applications: Record<string, Application>,
 ) {
-  const appConfig = applications[appId];
+  let appConfig = applications[appId];
   if (!appConfig) {
-    throw new Error(`Unknown application '${appId}'`);
+    //throw new Error(`Unknown application toto '${appId}'`);
+    // by default in local we add dedicated app
+    // if the app is not registred in 2api
+    appConfig = applications['dedicated'];
```
Comment (anooparveti) [inline packages/components/ovh-shell/src/client/index.ts:68]: **question:** This seems like a temporary change. Defaulting the app dedicated can be avoided.

File: .eslintrc.js (hunk 6 above)
Comment (chipp972) [reply .eslintrc.js:44]: This rule is turned off exclusively for typescript files.

As it is declared, it forces any js/ts file to declare a default export. But it is not always useful and most of the time one or multiple named export is better to stay consistent across the project. So I don't see why we force a default export in every file.

File: .eslintignore (hunk 5 above)
Comment (chipp972) [reply .eslintignore:5]: I think it was to avoid having too many changes in the PR and diluting the features added (the generator).
We can fix the eslint errors in another PR or in this one if you prefer.

File: .prettierignore [hunk 15]
```diff
@@ -0,0 +1,2 @@
+packages/manager/apps/container
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply .prettierignore:1]: Anoop the old .ts files in the core have not been prettier or lint in the past, so in the pull request we don't have to fix it, we can fix it on an other pull request

File: packages/manager/core/generator/app/templates/package.json.hbs [hunk 16]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "{{ packageName }}",
+  "version": "0.0.0",
+  "description": "{{ description }}",
+  "private": "true",
```
Comment (chipp972) [inline packages/manager/core/generator/app/templates/package.json.hbs:5]:
```suggestion
  "private": true,
```

Sinon la commande build à la racine bug

**Comment by cbourgois** [comment]
You squashed all commits from https://github.com/ovh/manager/tree/feat/core-dev-packages into 1 commit. Can you preserve @frenautvh and @Jisay as Co-author ?

File: packages/manager/core/generator/app/templates/package.json.hbs (hunk 16 above)
Comment (cbourgois) [reply packages/manager/core/generator/app/templates/package.json.hbs:5]: The reason is `private` is boolean, not a string.
https://docs.npmjs.com/cli/v9/configuring-npm/package-json#private

File: packages/components/ovh-shell/src/index.ts [hunk 17]
```diff
@@ -1,7 +1,7 @@
 import DirectClientMessageBus from './message-bus/direct-client';
 import IFrameMessageBus from './message-bus/iframe';
 
-export { default as useShellClient } from './client';
+export { default as initShellClient } from './client';
```
Comment (cbourgois) [inline packages/components/ovh-shell/src/index.ts:4]: Seems to be a Breaking Change but here is no BREAKING CHANGE commit.

File: docs/docs/guide/micro-applications.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (cbourgois) [reply docs/docs/guide/micro-applications.md]: GitHub mark it as Outdated on my side, but seems not.

File: docs/docs/guide/micro-applications.md [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+- [React](https://reactjs.org/docs/getting-started.html) - The library for web and native user interfaces
+- [Typescript](https://www.typescriptlang.org/) - Javascript with syntax for types
+
+The applications can be started for a specific region.
```
[the remark this answers is not in the document]
Comment (chipp972) [reply docs/docs/guide/micro-applications.md:10]: Removed the duplicated parts

[the remark this answers is not in the document]
Comment (cbourgois) [reply docs/docs/guide/micro-applications.md:10]: Seems 90% of https://github.com/ovh/manager/blob/feat/core-dev-packages-bis/docs/docs/guide/micro-applications.md are still a duplicate of https://github.com/ovh/manager/blob/feat/core-dev-packages-bis/docs/docs/guide/micro-applications.md.

The definition of micro application in this file is also wrong: applications like hub/dedicated/public-cloud/web/telecom are already micro-applications, written in AngularJS.
Others applications are standalone AngularJS applications, previously planned to be moved into AngularJS micro applications

Micro applications principle is not related to a specific technology/framework.

File: package.json [hunk 19]
```diff
@@ -23,12 +23,15 @@
     "format:css": "yarn run lint:css --fix",
     "format:html": "yarn run lint:html && prettier --parser=html --write \"packages/**/*.html\"",
     "format:js": "yarn run lint:js --fix",
+    "format:tsx": "yarn run lint:tsx --fix && prettier --write \"packages/manager/apps/**/*.{tsx,ts}\"",
     "format:md": "yarn run lint:md --fix",
     "generate:app": "sao ./packages/manager/tools/sao-ovh-manager-app",
+    "generate:uapp": "yarn workspace @ovh-ux/manager-generator run start",
```
[the remark this answers is not in the document]
Comment (chipp972) [reply package.json:28]: The libraries used to build it

File: docs/docs/guide/micro-applications.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (chipp972) [reply docs/docs/guide/micro-applications.md]: These are called Applications not micro-applications. There is a doc file called applications.md that already talks about them.

File: .eslintrc.js (hunk 6 above)
Comment (anooparveti) [reply .eslintrc.js:44]: I think we can have uniformity and prefer to have default export from all the files. @chipp972

Comment (chipp972) [reply .eslintrc.js]: Maybe uniformity is good when we don't have types and rely on naming convention and uniform file structure. But in Typescript, it is not very relevant and encourages bad behaviors like renaming a component across the project.

But ok I put back the rule

File: package.json (hunk 19 above)
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply package.json:28]: I didn't work on sao ovh project but it seem to generate a standalone app in angular
In the project migration, we named new applications react as 'uapp' 

maybe it is disturbing ?

File: packages/components/ovh-shell/src/index.ts (hunk 17 above)
Comment (aboungnaseng-ovhcloud) [reply packages/components/ovh-shell/src/index.ts:4]: Ok I updated the commit with a breaking change

File: docs/docs/guide/micro-applications.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (chipp972) [reply docs/docs/guide/micro-applications.md]: Removed the doc

File: packages/manager/core/application/src/ovh-routing.tsx [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+  ErrorElement: Element;
+  breadcrumb: () => unknown;
+};
+
+function HidePreloader(): JSX.Element {
+  const shell = useShell();
+  useEffect(() => {
+    shell.ux.hidePreloader();
+  }, []);
+  return undefined;
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-routing.tsx:34]: **question:** Can you return `null` instead?

File: packages/manager/core/application/src/ovh-routing.tsx [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+  const location = useLocation();
+
+  const shell = useShell();
+  useEffect(() => {
+    shell.routing.stopListenForHashChange();
+  }, []);
+  useEffect(() => {
+    shell.routing.onHashChange();
+  }, [location]);
+  return undefined;
```
Comment (anooparveti) [inline packages/manager/core/application/src/ovh-routing.tsx:47]: **question:** Can you return `null` instead?

File: packages/manager/core/generator/app/conditional-templates/dashboard/index.tsx.hbs [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+export default function Dashboard() {
+  const { t } = useTranslation('{{appName}}/dashboard');
+   const data = useLoaderData() as Record<string, unknown>;
+
+  return (
+    <div>
+      <Breadcrumb/>
+      <h1>{t('title')}</h1>
+      <div>Dashboard page</div>
```
Comment (anooparveti) [inline packages/manager/core/generator/app/conditional-templates/dashboard/index.tsx.hbs:34]: **nitpick:** Can be loaded from the translation files (packages/manager/core/generator/app/conditional-translations/dashboard/Messages_fr_FR.json)

File: packages/manager/core/application/src/ovh-routing.tsx (hunk 21 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/application/src/ovh-routing.tsx:47]: same answer below

File: packages/manager/core/generator/app/conditional-templates/dashboard/index.tsx.hbs (hunk 22 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/core/generator/app/conditional-templates/dashboard/index.tsx.hbs:34]: 👍

File: packages/components/ovh-shell/src/client/index.ts (hunk 14 above)
Comment (aboungnaseng-ovhcloud) [reply packages/components/ovh-shell/src/client/index.ts:68]: how can we avoided this ? we make this fix to run in local env

File: packages/manager/core/generator/app/conditional-templates/dashboard/index.tsx.hbs [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <div>
+      <Breadcrumb/>
+      <h1>{t('title')}</h1>
+      <Suspense fallback="">
+        <Await resolve={data.services}>
+          {(services) => {
+            const count = services.length;
+            if (count === 0) {
+              return <div>No services info</div>
```
Comment (anooparveti) [inline packages/manager/core/generator/app/conditional-templates/dashboard/index.tsx.hbs:39]: I think we need translation for `No Services info` as well

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by cbourgois — DISMISSED** [review]

**Review by cbourgois — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]
